Bump script bytecode version after token enum change

nikitalita opened the pull request:
See #67777 .

This change altered the token enum values when `ABSTRACT` was inserted, thus rendering all previous 4.x script bytecode incompatible with our parser. This necessitates bumping the bytecode version number.

nikitalita committed:
Bump script bytecode version after token enum change

RedMser commented:
There is a comment in `gdscript_tokenizer_buffer.cpp` which reads

> Unused, kept for compatibility. Please remove at next `TOKENIZER_VERSION` increment.

Since you are incrementing the version here, it would be a good idea to do this as a clean-up as well.

nikitalita commented:
> Since you are incrementing the version here, it would be a good idea to remove this as a clean-up as well.

Ok, I have removed it

RedMser commented on modules/gdscript/gdscript_tokenizer_buffer.cpp:
```diff
@@ -296,7 +296,7 @@ Vector<uint8_t> GDScriptTokenizerBuffer::parse_code_string(const String &p_code,
 	encode_uint32(constant_map.size(), &contents.write[4]);
 	encode_uint32(token_lines.size(), &contents.write[8]);
-	encode_uint32(0, &contents.write[12]); // Unused, kept for compatibility. Please remove at next `TOKENIZER_VERSION` increment.
+	encode_uint32(0, &contents.write[12]);
```
You just removed the comment. You should remove the whole line instead.

nikitalita replied:
Ok, I have done this.

nikitalita replied:
To be clear, do you want me to change the size of the header or leave the header the same size?

RedMser replied:
Hmm not sure, the code reading this probably needs adjustment as well. Someone who works on GDScript should be able to give better advice.

dalexeev replied:
Yes, you should resize the header and recalculate the offsets, in both encoding and decoding code. You should **not** replace `resize()` with `resize_zeroed()`, as it is less performant.

nikitalita commented:
@dalexeev  Ok, I implemented that as requested. I also changed it such that it checks for the exact bytecode version instead of just checking if it's newer, since we can't parse the content header of previous versions correctly anymore

nikitalita commented:
@dalexeev does this look good to you?
